Source/WebCore/rendering/RenderViewTransitionCapture.cpp

```diff
@@ -36,9 +36,41 @@ RenderViewTransitionCapture::RenderViewTransitionCapture(Type type, Document& do
     : RenderReplaced(type, document, WTFMove(style), { }, ReplacedFlag::IsViewTransitionCapture)
 { }
 
-void RenderViewTransitionCapture::setImage(const LayoutSize& size)
+void RenderViewTransitionCapture::setImage(RefPtr<ImageBuffer> oldImage, const LayoutSize& size)
 {
     setIntrinsicSize(size);
+    m_oldImage = oldImage;
+}
+
+void RenderViewTransitionCapture::paintReplaced(PaintInfo& paintInfo, const LayoutPoint& paintOffset)
+{
+    GraphicsContext& context = paintInfo.context();
+
+    LayoutRect contentBoxRect = this->contentBoxRect();
+
+    if (context.detectingContentfulPaint()) {
+        if (!context.contentfulPaintDetected() && m_oldImage)
+            context.setContentfulPaintDetected();
```

Comment:
Do view transitions really trigger contentful paint?

Reply:
They shouldn't. Maybe we should remove this code?